osquery/utils/aws/aws_util.cpp

```diff
@@ -427,7 +460,11 @@ bool isEc2Instance() {
       return; // Not EC2 instance
     }
 
+    auto token = getIMDSToken();
     http::Request req(kEc2MetadataUrl);
+    if (!token.empty()) {
```

Comment:
It's okay to proceed here with an empty `token`, if `getIMDSToken` was not successful?

Reply:
Already tried that :) 
It will result in `401 -- unauthorized`

Reply:
Sorry, what I meant was, this is the clause for "if the token is not empty" (then add it to the HTTP request), but what if it _is_ empty? It seems like execution just continues on line 468. Is that okay?

Reply:
Yes, that is intended! That is the fallback to v1 of the API. The only difference between `GET` calls of v1 and v2 is the presence of the IMDS token in the request header.